src/ProgressOnderwijsUtils/Collections/RootedTree.cs

```diff
@@ -44,6 +44,12 @@ public RootedTree<T> Parent
         public RootedTree<T> Root
             => PathSegments.Last().ThisSubTree.RootHere();
 
+        public RootedTree<T>? PreviousSibling()

```

Comment:
waarom nullable ipv met een maybe?

Reply:
Wat zou je dan als foutieve waarde willen returnen?

Reply:
We kunnen hier niet besluiten of de caller het goed/fout vind toch?

Reply:
Ik heb nullability zojuist weggehaald. Op deze manier moeten we altijd controleren op `HasValue` alvorens er iets mee te doen zoals op andere plekken in de tree.

Reply:
hoe bedoel je goed/fout, er is maybe een sibling

Reply:
Ja daar zit wat in. Voor nu heb ik t ontsloten zoals bij `Parent` gekeken wordt of hij een waarde heeft en dat is via `HasValue`. Indien wenselijk zouden we dat (zn geheel) naar maybe's kunnen omschrijven (en niet alleen next/previous sibling).

Reply:
er is mogelijk een sibling.  Er is mogelijk geen.  Je kunt "geen" representeren als maybe.error; maar ook als null - en ook als RootedTree, want de RootedTree is nu al een cursor die buiten de boom mag vallen; er is een `HasValue`.  `HasValue` houden en toch met maybe werken lijkt me iig niet handig; en het voordeel van maybe boven nullable zie ik ook niet; nullable zou daarentegen me wel iets van bij kunnen voorstellen - mits consequent doorgevoert, en dat is wel iets meer werk.

Reply:
Het `Maybe.Error` vs `Maybe.Ok` concept zorgde ervoor dat ik denk in goed/fout. Ik zou niet direct een andere naam/betere naam kunnen verzinnen. Dat er wel/niet een sibling is betekent niet direct dat 't goed (`Maybe.Ok`)/fout (`Maybe.Error`) is. Misschien is `Success`/ `Failed` een idee. Je voert een actie uit en dat slaagt of faalt in zn geheel. Met woorden als Error/Ok zeg je al iets over wat je er van vindt. Dat besluit zou niet bij de implementer moeten liggen, maar bij de caller naar mijn idee.

Reply:
ik zou dat niet zo letterlijk lezen. Die dingen hebben gewoon onhandige namen in onze code.

@EamonNerbonne sure. Het was ook een vraag; ik zag laatst ook een throw ipv een maybe, dus wou ff checken dat t bewust was.

Reply:
prima; Ik ben ook niet tegen een tree-api refresh, maar dan wel consequent ;-)